### PR TITLE
[ReportsConfig] Extract API

### PR DESCRIPTION
**Pull request by Guillar1**

### CODE REVIEW

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import { default as React_2 } from 'react';
+import type { Report } from '@itwin/insights-client';
+import { ReportsClient } from '@itwin/insights-client';
+import { StagePanelLocation } from '@itwin/appui-react';
+import { StagePanelSection } from '@itwin/appui-react';
+import type { UiItemsProvider } from '@itwin/appui-react';
+import type { Widget } from '@itwin/appui-react';
+
+// @public (undocumented)
+export class BulkExtractor {
```
Comment (MindaugasButkus) [inline packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:23]: I wonder if this should be exported. It's a pretty hacky solution for tracking running extractions.

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    onClickClose: () => void;
+    // (undocumented)
+    report: Report;
+}
+
+// @public
+export const Reports: ({ onClickAddReport, onClickReportModify, onClickReportTitle, }: ReportsProps) => JSX.Element | null;
+
+// @public (undocumented)
+export const REPORTS_CONFIG_BASE_URL = "https://api.bentley.com";
```
Comment (MindaugasButkus) [inline packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:69]: I didn't see the base url exposed for other widgets.

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md (hunk 1 above)
Comment (Guillar1) [reply packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:23]: It's out in the wild already. I can just mark it internal.

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md (hunk 2 above)
Comment (Guillar1) [reply packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:69]: G&M widget does not need it yet.. EC3 does not have it defined anywhere, you can define it on your own. I don't even think its worth exporting this. What do you think?

Comment (MindaugasButkus) [reply packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:69]: Yeah, I don't think it's needed. For now mark it as internal if it is already exposed.

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md (hunk 1 above)
Comment (Guillar1) [reply packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:23]: Ouch. 
image
I believe the reason it's there is to allow mocking of the BulkExtractor.

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md (hunk 2 above)
Comment (Guillar1) [reply packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:69]: Done.

**Review by MindaugasButkus — APPROVED** [review]

File: packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md (hunk 1 above)
Comment (MindaugasButkus) [reply packages/itwin/reports-config-widget/api/reports-config-widget-react.api.md:23]: Sounds like a bigger change than I expected initially. I think it's better to defer fixing this for now.

**Pull request merged**